src/core/dbus-unit.c

```diff
@@ -1839,6 +1839,26 @@ static int bus_unit_set_live_property(
                 return 1;
         }
 
+        /* Two boolean settings that only apply to active units. We don't actually write those to /run, this
+         * state is managed internally. */
+
+        if (STR_IN_SET(name, "NeedsRestart", "NeedsReload")) {
```

Comment:
hmm, this is quite asymmetric now... So far we tried hard to keep the real bus properties and those passed to SetUnitProperties() as much in sync as we can...

Either this should become also a "StateFlags" (or whatever better name one might find), or StateFlags should become two bools again. it's otherwise too confusing that "systemctl show -p NeedsRestart" fails, but "systemctl set-property NeedsRestart=" works, and vice versa for StateFlags.

This is user visible API after alll

Reply:
i'd probably still make this an enum